### PR TITLE
Add code to rotate polygon objects in 90-degree increments

### PR DESCRIPTION
**Pull request by aarograh**
CASL Ticket # - 6276

### CODE REVIEW

File: src/Geom_Poly.f90 [hunk 1]
```diff
@@ -2062,4 +2065,86 @@ PURE SUBROUTINE createArcFromQuad(thisPoly,iquad,circle)
   ENDIF
 ENDSUBROUTINE createArcFromQuad
 !
+!-------------------------------------------------------------------------------
+!> @brief Rotates a polygon type in 90-degree rotations in the x-y plane
+!> @param this the polygon to rotate
+!> @param nrotations the number of clockwise 90-degree rotations to apply
+!> @returns new the rotation polygon type
```
Comment (stimpsonsg) [inline src/Geom_Poly.f90]: rotation => rotated?

File: unit_tests/testGeom_Poly/testGeom_Poly.f90 [hunk 2]
```diff
@@ -2955,4 +2960,116 @@ SUBROUTINE testEquivalence()
...
+    ASSERT_APPROXEQ(new%quadEdge(2,i),-old%quadEdge(1,i),'%quadEdge, y')
+    ASSERT_APPROXEQ(new%quadEdge(3,i),old%quadEdge(3,i),'%quadEdge, z')
+  ENDDO !i
+
+  !Cleanup
+  CALL testGraph%clear()
+  CALL old%clear()
+  CALL new%clear()
+
+ENDSUBROUTINE testRotateClockwise
```
Comment (stimpsonsg) [inline unit_tests/testGeom_Poly/testGeom_Poly.f90]: excellent unit test

File: src/Geom_Poly.f90 (hunk 1 above)
Comment (aarograh) [reply src/Geom_Poly.f90]: yeah, good catch.  fixed

**Comment by aarograh** [comment]
The last commit fixes a defect that wasn't caught initially.  All the rotation was done right, but for the quadratic edges, the final (x,y) values were assigned using the x value for both x and y.  Because my unit test used a simple triangle with no quadratic edges, this was missed.  I switched the unit test to use a polygon with quadratic edges to cover this bug.

There are some minor changes required on the MPACT side, so this needs to be held until that branch is ready as well.

**Pull request merged**